examples/maze-solver/maze.py

```diff
@@ -0,0 +1,204 @@
+# Copyright 2019 D-Wave Systems Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import print_function
+
+import dwavebinarycsp as dbc
+from re import match
+
+
+def build_bqm(n_rows, n_cols, start, end, walls, penalty_per_tile=0.5):
+    """Returns a BQM that corresponds to a valid path through a maze. This maze is described by the parameters.
+
+    Specifically, it uses the parameters to build a maze constraint satisfaction problem (CSP). This maze CSP is then
+    converted into the returned BQM.
+
+    Note: If penalty_per_tile is too large, the path will be too heavily penalized and the optimal solution might
+    produce no path at all.
+
+    Args:
+        n_rows: Integer. The number of rows in the maze.
+        n_cols: Integer. The number of cols in the maze.
+        start: String. The location of the starting point of the maze. String follows the format of get_label(..).
+        end: String. The location of the end point of the maze. String follows the format of get_label(..).
+        walls: List of Strings. The list of inner wall locations. Locations follow the format of get_label(..).
+        penalty_per_tile: A number. Penalty for each tile that is included in the path; encourages shorter paths.
+
+    Returns:
+        A dimod.BinaryQuadraticModel
+    """
+    maze = Maze(n_rows, n_cols, start, end, walls)
+    return maze.get_bqm(penalty_per_tile)
+
+
+def get_label(row, col, direction):
+    """Provides a string that follows a standard format for naming constraint variables in Maze.
+    Namely, "<row_index>,<column_index><north_or_west_direction>".
+
+    Args:
+        row: Integer. Index of the row.
+        col: Integer. Index of the column.
+        direction: String in the set {'n', 'w'}. 'n' indicates north and 'w' indicates west.
+    """
+    return "".join([str(row), ",", str(col), direction])
```

Comment:
Cute way of doing this (maybe also more readable):
```
return "{row},{col}{direction}".format(**locals())
```

Obviously, this is just a hack until world upgrades to Python 3.6+, when we'll be able to use interpolated strings for this:
```
return f"{row},{col}{direction}"
```
:smile:

Reply:
I really like this new line of code!